source/Mission.cpp

```diff
@@ -856,6 +860,10 @@ bool Mission::IsSatisfied(const PlayerInfo &player) const
 	for(const NPC &npc : npcs)
 		if(!npc.HasSucceeded(player.GetSystem()))
 			return false;
+	// All timers must be complete
+	for(const Timer &timer : timers)
+		if(!timer.IsComplete())
+			return false;
```

Comment:
Do we want to require this?

Reply:
Initially I was thinking yes, but it does make sense to allow for "optional" timers, so I will add an "optional" attribute that can be added to circumvent this check. (Thus still defaulting to the timer being required to complete the mission.)